Tests/Realm.Tests/Sync/UserManagementTests.cs

```diff
@@ -147,6 +147,35 @@ public void AppRemoveUser_RemovesUser()
             });
         }
 
+        [Test]
+        public void AppDeleteUserFromServer_RemovesUser()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var username = SyncTestHelpers.GetVerifiedUsername();
+                var password = SyncTestHelpers.DefaultPassword;
+                var user = await GetUserAsync(app: null, username, password);
+
+                Assert.That(DefaultApp.CurrentUser, Is.EqualTo(user));
+
+                await DefaultApp.DeleteUserFromServerAsync(user);
+                Assert.That(DefaultApp.CurrentUser, Is.Null);
+
+                Exception ex = null;
+                try
+                {
+                    await DefaultApp.LogInAsync(Credentials.EmailPassword(username, password));
+                }
+                catch (Exception e)
+                {
+                    ex = e;
+                }
+
+                Assert.That(ex, Is.TypeOf<AppException>());
+                Assert.That(ex.Message, Is.EqualTo("InvalidPassword: invalid username/password"));
```

Comment:
If you want to make it slightly shorter you can do something like:
```csharp
var ex = Assert.Throws<AppException>(() => await DefaultApp.Login....)
Assert.That(ex.Message, Is.EqualTo("InvalidPassword: invalid username/password"));

```
This way you don't need to write the try catch. Unfortunately the new style (Assert.That) does not allow to have the exception returned.

Reply:
I'm not sure this will work with await 😕

Reply:
It seems that there is the async version: https://docs.nunit.org/articles/nunit/writing-tests/assertions/classic-assertions/Assert.ThrowsAsync.html

Reply:
I tried the async version already. It never returns. That's why I went with that solution. Also I understood that we wanted to only use the constraint model, which I think it's a good decision. So that was one more reason not to dig deeper.